thread_pthread.c

```diff
@@ -687,9 +788,14 @@ native_thread_init(struct rb_native_thread *nt)
 void
 Init_native_thread(rb_thread_t *main_th)
 {
+    int r;
+    if ((r = pthread_rwlock_init(&rb_internal_thread_event_hooks_rw_lock, NULL))) {
```

Comment:
Thanks a lot for this contribution, and apologies if this question does not make any sense but...

I'm curious -- why is `pthread_rwlock_init` needed if `rb_internal_thread_event_hooks_rw_lock` if already initialized with static `PTHREAD_RWLOCK_INITIALIZER`? I was under the impression that you could use either to do initialization, but did not need to do both.

Reply:
Looks like you're right: https://www.daemon-systems.org/man/pthread_rwlock.3.html

> The macro PTHREAD_RWLOCK_INITIALIZER can be used to initialize a
     read/write lock when the allocation can be done statically, no error
     checking is required, and the default attributes are appropriate.  The
     behavior is similar to calling pthread_rwlock_init() with attr specified
     as NULL.

Reply:
https://github.com/ruby/ruby/pull/6083